fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Memo.java

```diff
@@ -215,22 +229,19 @@ private void mergeGroup(Group srcGroup, Group dstGroup) {
                     reinsertExpression.getGroup().replaceBestExpression(reinsertExpression, existGroupExpression);
                     // existingGroupExpression merge the state of groupExpression
                     existGroupExpression.mergeGroupExpression(reinsertExpression);
+                } else {
+                    reinsertExpression.getGroup().deleteBestExpression(reinsertExpression);
```

Comment:
reinsertExpression.getGroup() will be merged, why need to delete the best expression

Reply:
Because the best expression will also be merged, it need to be clear first